src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,44 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final Random random = new Random();
+    private static final int MAX_INT = 25;
+    private static final Figure DEF_CIRCLE = new Circle();
+    private static final ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        Figure[] figures = {new Rectangle(),
+                            new RightTriangle(),
+                            new Circle(),
+                            new Square(),
+                            new IsoscelesTrapezoid()};
+
+        Figure figure = figures[random.nextInt(figures.length)];
+        if (figure instanceof Rectangle) {
+            figure.setColor(colorSupplier.getRandomColor());
+            ((Rectangle) figure).setFirstLeg(random.nextInt(MAX_INT) + 1);
+            ((Rectangle) figure).setSecondLeg(random.nextInt(MAX_INT) + 1);
+        } else if (figure instanceof RightTriangle) {
+            ((RightTriangle) figure).setFirstLeg(random.nextInt(MAX_INT) + 1);
+            ((RightTriangle) figure).setSecondLeg(random.nextInt(MAX_INT) + 1);
+        } else if (figure instanceof IsoscelesTrapezoid) {
+            ((IsoscelesTrapezoid) figure).setFirstLeg(random.nextInt(MAX_INT) + 1);
+            ((IsoscelesTrapezoid) figure).setSecondLeg(random.nextInt(MAX_INT) + 1);
+            ((IsoscelesTrapezoid) figure).setThirdLeg(random.nextInt(MAX_INT) + 1);
+        } else if (figure instanceof Square) {
+            ((Square) figure).setFirstLeg(random.nextInt(MAX_INT) + 1);
+        } else if (figure instanceof Circle) {
+            ((Circle) figure).setRadius(random.nextInt(MAX_INT) + 1);
+        }
```

Comment:
This kind of logic is very difficult to understand. First of all, you don't need an enum. And second, all you need is to get a random number from 1 to 5 and use it to pick a figure.

Reply:
1. There is no enum;
2. this code get a random number from 1 to 5 and used it to pick a figure 

Figure[] figures = {new Rectangle(),
                            new RightTriangle(),
                            new Circle(),
                            new Square(),
                            new IsoscelesTrapezoid()};

        Figure figure = figures[random.nextInt(figures.length)];

Reply:
Rewrited the logic of figure creating